src/main/java/school/faang/user_service/service/user/UserService.java

```diff
@@ -25,4 +35,19 @@ public UserDto getUser(long userId) {
     public List<UserDto> getUsersByIds(List<Long> ids) {
         return userMapper.toDto(userRepository.findAllById(ids));
     }
+
+    @Transactional
+    public void deactivate(long userId) {
+        userValidator.userExistenceInRepo(userId);
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new UserNotFoundException(MessageError.USER_NOT_FOUND_EXCEPTION));
+        user.setActive(false);
+        userRepository.save(user);
```

Comment:
you can omit saving user in repo here, because that is unnecessary when you use @Transactional